koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -945,3 +954,271 @@ def handler(self, src, target, opts=None):
             'repositories': all_repositories,
             'koji_builds': all_koji_builds,
         }
+
+
+class BuildSourceContainerTask(BaseContainerTask):
+    # Start builds via osbs
+    Methods = ['buildSourceContainer']
+    # Same value as for regular 'build' method.
+    _taskWeight = 2.0
+
+    # JSON Schema definition for koji buildSourceContainer task parameters
+    # Used to validate arguments passed to the handler() method of this class
+    PARAMS_SCHEMA = {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "description": "Parameters for a koji buildSourceContainer task.",
+
+        "type": "array",
+        "items": [
+            {
+                "type": "string",
+                "description": "Build target."
+            },
+            {
+                "type": ["object"],
+                "properties": {
+                    "scratch": {
+                        "type": "boolean",
+                        "description": "Perform a scratch build?"
+                    },
+                    "koji_build_id": {
+                        "type": ["integer"],
+                        "description": "Koji build id for sources"
+                    },
+                    "koji_build_nvr": {
+                        "type": ["string"],
+                        "description": "Koji build nvr for sources"
+                    },
+                    "signing_intent": {
+                        "type": ["string"],
+                        "description": "Signing intent of the ODCS composes."
+                    }
+                },
+                "oneOf": [{
+                    "required": ["koji_build_nvr"]
+                }, {
+                    "required": ["koji_build_id"]
+                }],
+                "additionalProperties": "false"
+            }
+        ],
+        'minItems': 2
+    }
+
+    def __init__(self, id, method, params, session, options, workdir=None, demux=False):
+        BaseContainerTask.__init__(self, id, method, params, session, options,
+                                   workdir)
+        self._osbs = None
+        self.demux = demux
+
+        self._log_handler_added = False
+
+    def runBuilds(self, target_info, scratch=False, component=None, koji_build_id=None,
+                  koji_build_nvr=None, signing_intent=None):
+        self.logger.debug("Spawning job for sources")
+
+        kwargs = dict(
+            target_info=target_info,
+            scratch=scratch,
+            component=component,
+            koji_build_id=koji_build_id,
+            koji_build_nvr=koji_build_nvr,
+            signing_intent=signing_intent,
+        )
+
+        results = []
+        semi_results = self.createSourceContainer(**kwargs)
+        if semi_results is not None:
+            results = [semi_results]
+
+        self.logger.debug("Results: %r", results)
+        return results
```

Comment:
Is there any reason to have this method? It basically just calls `createSourceContainer()`.

Reply:
right I can remove it and call it directly in handler